### PR TITLE
Enable nullable annotations, round 2 of x

### PR DESCRIPTION
**Pull request by iamcarbon**
Round 2

### CODE REVIEW

**Comment by xoofx** [comment]
I had to merge another PR in the meantime and there are a few merge conflicts, could you have a look?

File: src/Markdig/Extensions/Tables/PipeTableParser.cs [hunk 1]
```diff
@@ -59,7 +61,7 @@ public override bool Match(InlineProcessor processor, ref StringSlice slice)
             var position = processor.GetSourcePosition(slice.Start, out int globalLineIndex, out int column);
             var localLineIndex = globalLineIndex - processor.LineIndex;
 
-            if (tableState == null)
+            if (tableState is null)
```
Comment (MihaZupan) [inline src/Markdig/Extensions/Tables/PipeTableParser.cs:64]: I don't mind these changes (personally prefer is/is not), but it doesn't make a difference for nullable analyzers right?

File: src/Markdig/Parsers/ParserList.cs [hunk 2]
```diff
@@ -60,11 +61,11 @@ protected ParserList(IEnumerable<T> parsersArg) : base(parsersArg)
             var tempCharMap = new Dictionary<char, T[]>();
             foreach (var parser in this)
             {
-                if (parser.OpeningCharacters != null && parser.OpeningCharacters.Length != 0)
+                if (parser.OpeningCharacters is { Length: > 0 })
```
Comment (MihaZupan) [inline src/Markdig/Parsers/ParserList.cs:64]: We should stay consistent with the type of checks. The same pattern is used above in the same method but wasn't changed.

**Comment by MihaZupan** [comment]
Thanks, I think this is great for Markdig. Even if we might end up with a ton of `!`s in code rn, it's a great start.

File: src/Markdig/Parsers/ParserList.cs (hunk 2 above)
Comment (iamcarbon) [reply src/Markdig/Parsers/ParserList.cs:64]: Agreed. I had planned to follow up to apply these changes everywhere, but didn't want this PR to become to big. I replaced the remaining instances of this pattern to address this feedback now.

File: src/Markdig/Extensions/Tables/PipeTableParser.cs (hunk 1 above)
Comment (iamcarbon) [reply src/Markdig/Extensions/Tables/PipeTableParser.cs:64]: The analyzers pick up both patterns, but is is slightly more correct as it bypasses the equality operator and prevents use on structs.  I'd like to followup with another PR after this to replace the remaining == null with is null / for consistency.

**Comment by iamcarbon** [comment]
Notable, they'll likely be at least one remaining round to enable the next batch of files. After that, we should be able to enable nullable at the project level, and remove the inline #nullable enables.

**Comment by xoofx** [comment]
Thanks!

**Pull request merged**